### ISSUE TITLE
Msbuild gets stuck in net7.0 , preventing further builds until reboot

### ISSUE THREAD

**Issue by Blackclaws** [dotnet/msbuild#8010]
### Issue Description
The issue appears when stopping a build that is currently being done in the wrong spot. The console prints: "attempting to cancel the build". However msbuild just gets stuck.

### Steps to Reproduce
I'm not quite sure how to get into this state. I have had success running multiple dotnet builds in parallel and then cancelling some of them.

### Expected Behavior
```dotnet build``` and other commands keep working.

### Actual Behavior
Either ```dotnet build``` stops working and just reports:

```
Build Failed.
0 Warnings 0 Errors
```
or if you kill all running dotnet processes you can now build again but the build just gets stuck:
```
dotnet build
MSBuild version 17.4.0-preview-22428-01+14c24b2d3 for .NET
^CAttempting to cancel the build...
Attempting to cancel the build...
^C^C
```

### Analysis
I am guessing this has to do with the central build server that was introduced by net 7.0

### Versions & Configurations
dotnet version: 7.0.100-rc.1.22431.12

**Comment by rainersigwald** [comment]
Definitely sounds MSBuild-server related. If you see this again do you think you could capture memory dumps of the existing processes?

cc @rokonec @AR-May @MichalPavlik

**Comment by Blackclaws** [comment]
If you give me quick rundown of how to find the correct process to dump memory for and where to upload it I'll do that.

**Comment by rainersigwald** [comment]
You can use [`dotnet-dump`](https://learn.microsoft.com/dotnet/core/diagnostics/dotnet-dump) to explore processes:

`dotnet-dump ps` will give you a list of .NET processes running. Look through that list for the one running `dotnet build` and the ones with `MSBuild.dll` in the command line (MSBuild's worker processes).

Then you should be able to get memory dumps for each process with `dotnet-dump collect --process-id 59340` (insert your own PIDs).

Note that memory dumps may contain information you don't want to share in public, depending on your machine environment and your projects. If you'd like only Microsoft employees to see your dump, you can [open a Visual Studio feedback ticket](https://docs.microsoft.com/visualstudio/ide/feedback-options) and attach the .dmp files in private.

**Comment by Blackclaws** [comment]
That indeed sounds like it might be the core issue. If the issue pops up again I'll open a new issue.
